services/blockchain-connector/shared/sdk/encoder.js

```diff
@@ -69,7 +71,22 @@ const encodeBlock = (block) => {
 	return blockBuffer.toString('hex');
 };
 
+const encodeEvent = async (event) => {
+	const schemas = await getSchemas();
+	const eventSchema = schemas.event;
+
+	const schemaCompliantEvent = {
+		...event,
+		data: Buffer.from(event.data, 'utf8'),
+		topics: event.topics.map(t => Buffer.from(t, 'utf8')),
+	};
```

Comment:
Can we not use the `parseInputBySchema` utility to dynamically generate this?

Reply:
applied.